api/v1beta1/types_class.go

```diff
@@ -187,6 +187,10 @@ type AzureManagedControlPlaneClassSpec struct {
 	// DisableLocalAccounts disables getting static credentials for this cluster when set. Expected to only be used for AAD clusters.
 	// +optional
 	DisableLocalAccounts *bool `json:"disableLocalAccounts,omitempty"`
+
+	// FleetsMember is the spec for the fleet this cluster is a member of.
+	// +optional
+	FleetsMember *FleetsMember `json:"fleetManager,omitempty"`
```

Comment:
I don't think it would make sense to have several fleetmembers with the same name, as that would make managing them from the fleet manager really hard. I'm not sure if that is even allowed

Reply:
Right, is that a reason that the name should not be a part of the template type then?

Reply:
Yes, since we want the name to be unique to each cluster.

Reply:
Isn't the name part of the template type currently?
https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/babf7268fe5b6a4889d582ce34c8825e6bf97d59/config/crd/bases/infrastructure.cluster.x-k8s.io_azuremanagedcontrolplanetemplates.yaml#L270-L272

Reply:
Also unrelated: 
```suggestion
	FleetsMember *FleetsMember `json:"fleetsMember,omitempty"`
```

Reply:
> Isn't the name part of the template type currently?

I don't think so. The name is defined outside of the `FleetsMemberClassSpec` which contains the shared fields. A similar pattern is used for the `ManagedControlPlaneVirtualNetwork`: https://github.com/willie-yao/cluster-api-provider-azure/blob/aks-fleets-member/api/v1beta1/azuremanagedcontrolplane_types.go#L221

Reply:
Following the vnet pattern then, shouldn't this be using the class type?
```suggestion
	FleetsMember *FleetsMemberClassSpec `json:"fleetManager,omitempty"`
```

Reply:
@willie-yao Should we still address this?

Reply:
> @willie-yao Should we still address this?

I would vote "yes"

Reply:
@nojnhuh  Sorry, lost track of this thread. Will address this now!

Reply:
Hold on.. I'm a bit confused. The vnet pattern does this:
```
VirtualNetwork ManagedControlPlaneVirtualNetwork `json:"virtualNetwork,omitempty"`
```
Not
```
VirtualNetwork ManagedControlPlaneVirtualNetworkClassSpec `json:"virtualNetwork,omitempty"`
```

I think I might be misunderstanding something here.

Reply:
In the example you listed, `NetworkTemplateSpec` still contains fields that need to be unique such as subnets: https://github.com/willie-yao/cluster-api-provider-azure/blob/9f63c1d08fd9846504c1dc5ddf14143dd989bbef/config/crd/bases/infrastructure.cluster.x-k8s.io_azureclustertemplates.yaml#L574

This is the same pattern that I'm following as it has the shared fields inside a [`ClassSpec` field](https://github.com/willie-yao/cluster-api-provider-azure/blob/9f63c1d08fd9846504c1dc5ddf14143dd989bbef/api/v1beta1/types_template.go#L56) with the rest of the unique fields outside of it. 

I think I'm confused what the difference is between that and how the [FleetsMember struct](https://github.com/willie-yao/cluster-api-provider-azure/blob/aks-fleets-member/api/v1beta1/types.go#L1004) is set up

Reply:
Maybe this is something that needs to be changed across all ClusterClass types? Since fields like `subnet` shouldn't be showing up in the azure cluster template at all

Reply:
Ohh that makes sense.. Apologies for the confusion and thanks for the explanation! Will make this change now.